### PR TITLE
Add and structure docs

### LINKED ISSUE #36: Add Material For MkDocs documentation
This looks sexy and quite easy:

https://squidfunk.github.io/mkdocs-material/

No automagic like with sphinx, but since our package is so small then this should be enough.

For reference Reference
https://github.com/proofit404/stories

### PR DESCRIPTION
**Pull request by ChameleonTartu**

### CODE REVIEW

File: docs/contrib/tests.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+We use `pytest`, `flake8` and `mypy` as a quality gate. We follow `wemake_python_styleguide` to enforce quality.
```
Comment (thomasborgen) [inline docs/contrib/tests.md]: make `wemake_python_styleguide` a link to wemake site would be nice.
`https://wemake-python-stylegui.de/en/latest/`

File: docs/contrib/tests.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Since storage bucket names must be globally unique, we use a environment variable for the storage bucket name in tests. which means that you must set the env var `STORAGE_BUCKET_NAME` to whatever the name of your bucket is.
+>This might be removed in the future.
```
Comment (thomasborgen) [inline docs/contrib/tests.md]: Just remove this, we no longer use this env var.

File: docs/contrib/tests.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+poetry run flake8 .
+```
+
+To check typing:
+
+```
+poetry run mypy .
+```
+
+Default installation of virtual environment is in `.venv`. If you need customize parameters you can do it in `setup.cfg`.
```
Comment (thomasborgen) [inline docs/contrib/tests.md]: What is meant with 'if you need customize parameters you can do it in `setup.cfg`?
if people want to install the virtualenv in the project folder it should be in .venv only.
if people want to install other places then thats a thing poetry will handle so i don't think we need to care about it.
The setting `virtualenvs.in-project=false` should make poetry install it in some local poetry folder and activates it automatically.
So I'm for deleting this line.

File: docs/faq.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# FAQ
+
+## How is it different from the google-cloud-storage library designed and maintained by Google?
+
+This library is built on top of [google-cloud-storage](google-cloud-storage). `storage-bucket` library allows to work in functional style and abstracts away common use-cases which bloats code with boileraplate. It enables smooth integration with [`returns`](https://github.com/dry-python/returns) library functional style and powered by it.
```
Comment (thomasborgen) [inline docs/faq.md]: and `is` powered by it.
otherwise awesome paragraph

File: docs/faq.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+## What if there isn't feature I need?
```
Comment (thomasborgen) [inline docs/faq.md]: change to: `Missing a feature?`

File: docs/usage.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+# Normal way, this might throw exception... handle them yourself.
+my_data = download_file(
+    'my_bucket',
+    'my_file.txt',
+)
+use_data_for_something(my_data)
+
+
+# Returns Modal way
```
Comment (thomasborgen) [inline docs/usage.md]: `# Functional way` is better i think.

**Comment by ChameleonTartu** [comment]
@thomasborgen I updated docs and pushed then to `gh-pages`.

**Review by thomasborgen — APPROVED** [review]

**Pull request merged**